api/envoy/config/route/v3/route_components.proto

```diff
@@ -1130,13 +1151,22 @@ message RetryPolicy {
   // HTTP status codes that should trigger a retry in addition to those specified by retry_on.
   repeated uint32 retriable_status_codes = 7;
 
-  // Specifies parameters that control retry back off. This parameter is optional, in which case the
+  // Specifies parameters that control exponential retry back off. This parameter is optional, in which case the
   // default base interval is 25 milliseconds or, if set, the current value of the
   // `upstream.base_retry_backoff_ms` runtime parameter. The default maximum interval is 10 times
   // the base interval. The documentation for :ref:`config_http_filters_router_x-envoy-max-retries`
   // describes Envoy's back-off algorithm.
   RetryBackOff retry_back_off = 8;
 
+  // Specifies parameters that control a retry back off based on rate limited
+  // responses. When a request is rate limited by the server the server may
+  // include a header like ``Retry-After`` or ``X-RateLimit-Reset`` to provide
+  // feedback to the client on how long to wait before retrying. If configured,
+  // the rate limited retry back off strategy will be used instead of the
+  // default exponential back off strategy (configured using `retry_back_off`)
+  // whenever a response includes the matching headers.
+  RateLimitedRetryBackOff rate_limited_retry_back_off = 11;
```

Comment:
What do you think about folding this in to the `RetryBackoff` message? It seems logically related since it will govern what the final backoff time is?

Reply:
That's definitely possible. I figured it made more sense to make it clear that it's a separate code path with its own parameters and the new stat `upstream_rq_retry_backoff_ratelimited` to reflect whether it's being used or not.

In the doc update I pushed the mechanism does take a fair bit of explaining, so I think that might be another reason to keep them separated.